packages/admin-web/src/components/StepperPage.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import TextField from '@material-ui/core/TextField';
+import { SimpleStepper, SimpleStepperStep } from '@backstage/core-components';
+import {
+  Button,
+  FormControlLabel,
+  Radio,
+  RadioGroup,
+  Typography,
+  List,
+  ListItem,
+} from '@material-ui/core';
+
+const StepperPage = () => {
+  const [required, setRequired] = useState<boolean>(false);
+
+  const [oauthMethod, setOauthMethod] = useState<string>('');
+  const [clientId, setClientId] = useState<string>('');
+  const [clientSecret, setClientSecret] = useState<string>('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setOauthMethod((event.target as HTMLInputElement).value);
+    return setRequired(!!event.target.value);
+  };
+
+  const handleClientId = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClientId((event.target as HTMLInputElement).value);
+    return setRequired(event.target.value.length > 0);
+  };
+  const handleClientSecret = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClientSecret((event.target as HTMLInputElement).value);
+    return setRequired(event.target.value.length > 0);
+  };
+
+  const getStep1Content = () => {
```

Comment:
Maybe we should make transform these functions to React components? 

```jsx
<SimpleStepper>
          <SimpleStepperStep title="Step up Github Authentication">
              <OAuthSelector  />
            </SimpleStepperStep>
             <SimpleStepperStep title="Step up Github Authentication">
              <OAuthCredentials oauthMethod="oauth"  />
            </SimpleStepperStep>
  </SimpleStepperStep>
 ```
 
 Don't have a huge opinion on it though.